src/langsmith/prompt-engineering-quickstart.mdx

```diff
@@ -111,137 +249,110 @@ This guide will walk through how to create, test, and iterate on prompts using t
     });
     ```
+    </CodeGroup>
 
-        </CodeGroup>
+    This loads the prompt by name using `pull` for the latest committed version of the prompt that you're testing. You can also specify a specific commit by passing the commit hash `"<prompt-name>:<commit-hash>"`.
 
-    ### 4. Iterate on a prompt
+1. Run `test_prompt` :
 
-    LangSmith makes it easy to iterate on prompts with your entire team. Members of your workspace can select a prompt to iterate on, and once they are happy with their changes, they can simply save it as a new commit.
+    <CodeGroup>
 
-    To improve your prompts:
+    ```python Python
+    python test_prompt.py
+    ```
+
+    ```typescript TypeScript
+    npx tsx test_prompt.ts
+    ```
+
+    </CodeGroup>
 
-    * We recommend referencing the documentation provided by your model provider for best practices in prompt creation, such as [Best practices for prompt engineering with the OpenAI API](https://help.openai.com/en/articles/6654000-best-practices-for-prompt-engineering-with-the-openai-api) and [Gemini's Introduction to prompt design](https://ai.google.dev/gemini-api/docs/prompting-intro).
 
-    * To help with iterating on your prompts in LangSmith, we've created Prompt Canvas — an interactive tool to build and optimize your prompts. Learn about how to use [Prompt Canvas](/langsmith/observability-concepts#prompt-canvas).
+## 4. Iterate on a prompt
 
-    To add a new commit to a prompt, you can use the same [`push_prompt`](https://docs.smith.langchain.com/reference/python/client/langsmith.client.Client#langsmith.client.Client.push_prompt) (Python) or [`pushPrompt`](https://langsmith-docs-7jgx2bq8f-langchain.vercel.app/reference/js/classes/client.Client#pushprompt) (TypeScript) methods as when you first created the prompt.
+LangSmith makes prompt iteration collaborative and transparent. Members of your workspace can open an existing prompt, experiment with changes in the UI, and save those changes as a new commit on the same prompt, which preserves history for the whole team.
 
-        <CodeGroup>
+1. To create a new version of a prompt, call the same push method you used initially with the same prompt name and your updated template. LangSmith will record it as a new commit and preserve prior versions. You can add a tag to label the change.
+
+    Copy the following code to an `iterate_prompt` file:
+
+    <CodeGroup>
 
     ```python Python
     from langsmith import Client
     from langchain_core.prompts import ChatPromptTemplate
 
-    # Connect to the LangSmith client
     client = Client()
 
-    # Define the prompt to update
     new_prompt = ChatPromptTemplate([
         ("system", "You are a helpful chatbot. Respond in Spanish."),
         ("user", "{question}"),
     ])
 
-    # Push the updated prompt making sure to use the correct prompt name
-    # Tags can help you remember specific versions in your commit history
-    client.push_prompt("my-prompt", object=new_prompt, tags=["Spanish"])
```

Comment:
ah this actually creates a prompt tag, not a commit tag. i dont think we have a way to leave commit tags from the SDK. i'd remove this for now

Reply:
Ah, I see yeah, I ran it again and looked in the UI. Thanks for clarifying!